### PR TITLE
Inventory

### PR DESCRIPTION
**Pull request by karelhala**

### CODE REVIEW

File: src/SmartComponents/Inventory/Actions.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  }
+}
+
+Actions.propTypes = {
+  item: PropTypes.shape({
+    id: PropTypes.number
+  })
+};
+
+export default Actions;
```
Comment (martinpovolny) [inline src/SmartComponents/Inventory/Actions.js]: missing eol

File: src/redux/reducers/inventory/entities.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        entities: entities
+    }
+}
+
+function entitiesLoaded(state, { payload }) {
+    const entities = mergeArraysByKey([state.rows, payload]);
+    return {
+        ...state,
+        loaded: true,
+        rows: entities,
```
Comment (martinpovolny) [inline src/redux/reducers/inventory/entities.js]: @karelhala : why do you have `rows: entities` here and also `entities` below?

File: src/redux/reducers/inventory/entities.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        entities
+    }
+}
+
+function selectEntity(state, { payload: { id, selected } }) {
+    const ents = [...state.entities];
```
Comment (martinpovolny) [inline src/redux/reducers/inventory/entities.js]: would look better w/o the shortening ("ents")

File: src/redux/reducers/inventory/entities.js (hunk 2 above)
Comment (karelhala) [reply src/redux/reducers/inventory/entities.js]: Rows represent data receives from db, but entities represent filtered and/or sorted data in which connected component is really interested.

**Comment by ryelo** [comment]
@karelhala Can you write a doc to explain a little more about this as well as consumption?

**Comment by karelhala** [comment]
@ryelo sure thing!

**Comment by ryelo** [comment]
Awesome! Nice docs @karelhala

**Pull request merged**